orange/src/main/java/com/hartwig/hmftools/orange/algo/purple/LossOfHeterozygositySelector.java

```diff
@@ -34,63 +37,103 @@ final class LossOfHeterozygositySelector {
         return suspectGeneCopyNumbersWithLOH;
     }
 
-    private static boolean hasLOH(@NotNull GeneCopyNumber geneCopyNumber) {
-        return geneCopyNumber.minMinorAlleleCopyNumber() < 0.5 && geneCopyNumber.minCopyNumber() > 0.5;
+    private static boolean hasLOH(@NotNull GeneCopyNumber geneCopyNumber)
+    {
+        return geneCopyNumber.minMinorAlleleCopyNumber() < 0.5 && notFullyLostInTumor(geneCopyNumber);
     }
 
-    private static boolean hasReportedGermlineHetDeletion(@NotNull GeneCopyNumber geneCopyNumber,
-            @Nullable List<GermlineDeletion> allGermlineDeletions) {
-        if (allGermlineDeletions == null) {
-            return false;
-        }
+    private static boolean notFullyLostInTumor(@NotNull GeneCopyNumber geneCopyNumber)
+    {
+        return geneCopyNumber.minCopyNumber() > 0.5;
+    }
 
-        GermlineDeletion deletion = findByGene(allGermlineDeletions, geneCopyNumber.geneName());
-        if (deletion == null) {
+    private static boolean hasReportedGermlineHetDeletion(@NotNull GeneCopyNumber geneCopyNumber,
+            @Nullable List<GermlineDeletion> allGermlineDeletions)
+    {
+        if(allGermlineDeletions == null)
+        {
             return false;
         }
 
-        return deletion.Reported && deletion.TumorStatus == GermlineStatus.HET_DELETION;
+        return allGermlineDeletions.stream()
+                .anyMatch(d -> isMatchingReportedGermlineDeletion(d, geneCopyNumber.geneName(), GermlineStatus.HET_DELETION));
     }
 
-    @Nullable
-    private static GermlineDeletion findByGene(@NotNull List<GermlineDeletion> deletions, @NotNull String geneNameToFind) {
-        for (GermlineDeletion deletion : deletions) {
-            if (deletion.GeneName.equals(geneNameToFind)) {
-                return deletion;
-            }
+    private static boolean hasReportedGermlineHomDeletion(@NotNull GeneCopyNumber geneCopyNumber,
+            @Nullable List<GermlineDeletion> allGermlineDeletions)
+    {
+        if(allGermlineDeletions == null)
+        {
+            return false;
         }
-        return null;
+
+        return allGermlineDeletions.stream()
+                .anyMatch(d -> isMatchingReportedGermlineDeletion(d, geneCopyNumber.geneName(), GermlineStatus.HOM_DELETION));
     }
 
     @NotNull
-    private static GeneCopyNumber correctForGermlineImpact(@NotNull GeneCopyNumber geneCopyNumber) {
+    private static GeneCopyNumber correctForGermlineImpact(@NotNull GeneCopyNumber geneCopyNumber,
+            @NotNull List<GermlineDeletion> allGermlineDeletions)
+    {
+        double adjustedMinCopyNumber = adjustMinCopyNumberForGermlineImpact(geneCopyNumber, allGermlineDeletions);
         return ImmutableGeneCopyNumber.builder()
                 .from(geneCopyNumber)
-                .minCopyNumber(geneCopyNumber.minCopyNumber() - geneCopyNumber.minMinorAlleleCopyNumber())
+                .minCopyNumber(adjustedMinCopyNumber)
                 .minMinorAlleleCopyNumber(0D)
```

Comment:
It was overwritten to 0 already. Unchanged the value would be at least 0.5, otherwise the "hasLOH" check would have caused the LOH call, not the heterozygous germline deletion(s). So if we don't change the value at all, the adjusted GeneCopyNumber object would not actually represent an LOH region.
I think setting it to 0 is the best choice, since if there is an actual germline deletion, then it should theoretically be 0.    Wrongly aligned reads etc. could change that in practice, but I'm not sure how we would calculate a value that takes that into account. And I also think that we would just be setting ourselves up for failure in the sense that the resulting value could end up greater than 0.5, which would just needlessly create a problem for ourselves.